ddl/column.go

```diff
@@ -822,7 +822,7 @@ func doReorgWorkForModifyColumn(w *worker, d *ddlCtx, t *meta.Meta, job *model.J
 		return false, ver, errors.Trace(err)
 	}
 	reorgInfo, err := getReorgInfo(d.jobContext(job.ID), d, rh, job, dbInfo, tbl, BuildElements(changingCol, changingIdxs), false)
-	if err != nil || reorgInfo.first {
+	if err != nil || reorgInfo == nil || reorgInfo.first {
```

Comment:
Why modify column need this logic？

Reply:
Because `reorgInfo` is possible to be `nil` after the changes in `getReorgInfo`. `reorgInfo == nil` means the job is cancelling.